src/writer.rs

```diff
@@ -844,6 +853,54 @@ mod write_vtk_impl {
             let buf = IOBuffer::from(data);
             self.write_buf::<BO>(buf)
         }
+        fn write_color_scalars_buf<BO: ByteOrder>(&mut self, buf: IOBuffer) -> Result {
+            fn write_buf_impl<T, W, F>(vec: Vec<T>, writer: &mut W, convert_to_u8: F) -> Result
+            where
+                W: WriteBytesExt,
+                F: Fn(T) -> u8,
+            {
+                for elem in vec {
+                    W::write_u8(writer, convert_to_u8(elem))?;
+                }
+                Ok(())
+            }
+            fn convert_float<T: Into<f64>>(input: T) -> u8 {
+                (input.into() * 255.0).round().clamp(0.0, 255.0) as u8
+            }
+
+            match buf {
+                IOBuffer::Bit(v) => write_buf_impl(v, &mut self.0, |x| x * 255 as u8)?,
```

Comment:
Although `Bit` has the same content as `U8` we multiply by 255 here. Is that needed? The other cases are handled coherently I think, but this sticks out a little bit.